src/SemiFungible1155.sol

```diff
@@ -43,44 +60,55 @@ contract SemiFungible1155 is Upgradeable1155 {
     /// ENJIN EXAMPLE IMPLEMENTATION
     // Only to make code clearer. Should not be functions
     // TODO cleanup functions
+    /// @dev Identify if token at `_id` is non-fungible.
+    /// @dev Non-fungible tokens are used to represent the base type for hypercert tokens
     function isNonFungible(uint256 _id) internal pure returns (bool) {
         return _id & TYPE_NF_BIT == TYPE_NF_BIT;
     }
 
+    /// @dev Identify if token at `_id` is fungible.
+    /// @dev Fungible tokens are used to represent (fractional) ownership of a claim
     function isFungible(uint256 _id) internal pure returns (bool) {
         return _id & TYPE_NF_BIT == 0;
     }
 
+    /// @dev Get index of fractional token as `_id`
     function getNonFungibleIndex(uint256 _id) internal pure returns (uint256) {
         return _id & NF_INDEX_MASK;
     }
 
+    /// @dev Get base type ID for token at `_id`
     function getNonFungibleBaseType(uint256 _id) internal pure returns (uint256) {
         return _id & TYPE_MASK;
     }
 
+    /// @dev Identify that token at `_id` is base type.
+    /// @dev Upper 128 bits identify base type ID, lower bits should be 0
     function isNonFungibleBaseType(uint256 _id) internal pure returns (bool) {
         // A base type has the NF bit but does not have an index.
         return (_id & TYPE_NF_BIT == TYPE_NF_BIT) && (_id & NF_INDEX_MASK == 0);
     }
 
+    /// @dev Identify that token at `_id` is fraction of a claim.
+    /// @dev Upper 128 bits identify base type ID, lower bits should be > 0
     function isNonFungibleItem(uint256 _id) internal pure returns (bool) {
-        // A base type has the NF bit but does has an index.
         return (_id & TYPE_NF_BIT == TYPE_NF_BIT) && (_id & NF_INDEX_MASK != 0);
     }
 
     /// READ
 
-    function balanceOf(address _owner, uint256 _tokenID) public view override returns (uint256 tokenUserBalance) {
-        tokenUserBalance = tokenUserBalances[_tokenID][_owner];
+    function totalUnits(uint256 _typeID) external view returns (uint256 total) {
+        total = tokenValues[_typeID];
```

Comment:
What if someone passes in a tokenId?
Should we apply the Type mask?

Reply:
This method can actually go since I've added the `unitsOf(uint256 tokenID)` and `unitsOf(address owner, uint256 tokenID)`.

Reply:
done